routers/web/repo/editor.go

```diff
@@ -81,7 +84,11 @@ func editFile(ctx *context.Context, isNewFile bool) {
 		return
 	}
 
-	treeNames, treePaths := getParentTreeFields(ctx.Repo.TreePath)
+	// Check if the filename (and additional path) is specified in the querystring
+	// (filename is a misnomer, but kept for compatibility with Github)
+	filePath, fileName := path.Split(ctx.Req.URL.Query().Get("filename"))
```

Comment:
```suggestion
	filePath, fileName := path.Split(ctx.Query("filename"))
```

Reply:
This is incorrect. `ctx.Query` refers to `Form`, not `URL.Query()`.

Reply:
(hence I updated the comment to be more clear earlier)

Reply:
Original docs, before the migration from Macaron to Chi: https://pkg.go.dev/gopkg.in/macaron.v1?utm_source=godoc#Context.Query

Reply:
hmm if it not works we might considder to refactor it :D

Reply:
It does work. Perfectly.